test/e2e/util.go

```diff
@@ -0,0 +1,284 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func waitForDeploymentStatusCondition(t *testing.T, cl client.Client, timeout time.Duration, deploymentName types.NamespacedName, conditions ...appsv1.DeploymentCondition) error {
+	t.Helper()
+
+	return wait.PollImmediate(10*time.Second, timeout, func() (bool, error) {
+		dep := &appsv1.Deployment{}
+		if err := cl.Get(context.TODO(), deploymentName, dep); err != nil {
+			t.Logf("failed to get deployment %s: %v (retrying)", deploymentName.Name, err)
+			return false, nil
+		}
+
+		expected := deploymentConditionMap(conditions...)
+		current := deploymentConditionMap(dep.Status.Conditions...)
+		return conditionsMatchExpected(expected, current), nil
+	})
+}
+
+func getIngress(t *testing.T, cl client.Client, timeout time.Duration, ingressName types.NamespacedName) (string, error) {
+	t.Helper()
+	var address string
+	return address, wait.PollImmediate(10*time.Second, timeout, func() (bool, error) {
+		ing := &networkingv1.Ingress{}
+		if err := cl.Get(context.TODO(), ingressName, ing); err != nil {
+			t.Logf("failed to get deployment %s: %v (retrying)", ingressName.Name, err)
+			return false, nil
+		}
+		if len(ing.Status.LoadBalancer.Ingress) <= 0 || len(ing.Status.LoadBalancer.Ingress[0].Hostname) <= 0 {
+			return false, nil
+		}
+		address = ing.Status.LoadBalancer.Ingress[0].Hostname
+		return true, nil
+	})
+}
+
+func waitForDeletion(t *testing.T, cl client.Client, obj client.Object, timeout time.Duration) {
+	t.Helper()
+	deletionPolicy := v1.DeletePropagationForeground
+	_ = wait.PollImmediate(10*time.Second, timeout, func() (bool, error) {
```

Comment:
I still don't understand why do we have to do this. I think it's not blocking because the child resources don't have the `blockOwnerDeletion` set to true(based on my reading of the [docs](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#DeletionPropagation)).